### ISSUE TITLE
CI: Unit-10 RL shard red — MuZero_Train_UpdatesParameters (params don't update; residual after #1729)

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1752]
## Shard
`Tests (net10.0) - ModelFamily / Unit - 10 Regularization/RL/RAG2` — **1 of 1509** tests failing (everything else green).

## Failing test
`AiDotNet.Tests.UnitTests.ReinforcementLearning.ModelBasedAndMultiAgentTrainingTests.MuZero_Train_UpdatesParameters`
(`tests/AiDotNet.Tests/UnitTests/ReinforcementLearning/ModelBasedAndMultiAgentTrainingTests.cs:161`)

## Analysis
Reproduces locally in isolation (232 ms) — a **genuine model bug**, not a timeout/flake. The test trains a MuZero agent and asserts its parameters change; they don't move. This is a **residual after PR #1729** (which merged 2026-06-30 and *did* touch `src/ReinforcementLearning/Agents/MuZeroAgent.cs` + this test, but the assertion still fails on post-#1729 master, run 995b37b8d).

## Fix approach (fix the model, not the test)
Implement/repair the MuZero training step so a `Train` call actually updates the agent's parameters (representation / dynamics / prediction networks). Compare against the MuZero paper (Schrittwieser et al. 2020) — the loss must flow gradients to all three networks. Verify locally:
```
dotnet test tests/AiDotNet.Tests/AiDotNetTests.csproj --framework net10.0 -c Release \
  --filter "FullyQualifiedName~MuZero_Train_UpdatesParameters"
```

## Effort
Small–Medium. Self-contained (RL agent). No dependency on other shard work.

Part of the #1706 CI-shard greening effort.

**Comment by ooples** [comment]
**Root cause correction (actual, not the stale assertion hint):**

The test doesn't fail on "params didn't change" — `MuZeroAgent.Train()` **throws**:
> `System.InvalidOperationException : MuZeroAgent.Train currently builds one-step targets, but UnrollSteps is 5. Implement sequence replay / unrolled targets before configuring UnrollSteps > 1.`

`MuZeroOptions.UnrollSteps` defaults to **5** (paper value), but #1729 only implemented the one-step (K=1) unrolled targets and added a fail-fast guard for K>1. So the **out-of-box agent throws on `Train()`** (the test builds `MuZeroOptions` without setting `UnrollSteps`).

**Fix (being taken now):** default `UnrollSteps = 1` (the implemented capability) so the default agent actually trains with the correct one-step targets; a caller wanting the paper's K=5 sets it explicitly and still gets the clear guard exception. The one-step path trains all three networks (representation/dynamics/prediction) with proper targets — verified params update.

**Follow-up (separate enhancement, keep this issue's scope to the shard fix):** implement the full K-step unrolled training (sequence replay + per-unroll-step targets, Schrittwieser et al. 2020) so K>1 works and the default can return to 5.
